botocore/handlers.py

```diff
@@ -51,6 +52,7 @@
 # combination of uppercase letters, lowercase letters, numbers, periods
 # (.), hyphens (-), and underscores (_).
 VALID_BUCKET = re.compile('^[a-zA-Z0-9.\-_]{1,255}$')
+VERSION_ID_SUFFIX = re.compile(r'\?versionId=[^\s]+$')
```

Comment:
Should this allow for `?versionId=...` and `&versionId=...` (e.g., the query string contains values before `versionId`)?